modules/navigation/nav_map.cpp

```diff
@@ -103,9 +111,9 @@ void NavMap::set_link_connection_radius(real_t p_link_connection_radius) {
 }
 
 gd::PointKey NavMap::get_point_key(const Vector3 &p_pos) const {
-	const int x = static_cast<int>(Math::floor(p_pos.x / cell_size));
-	const int y = static_cast<int>(Math::floor(p_pos.y / cell_height));
-	const int z = static_cast<int>(Math::floor(p_pos.z / cell_size));
+	const int x = static_cast<int>(Math::round(p_pos.x / (cell_size * rasterizer_scale_factor)));
+	const int y = static_cast<int>(Math::round(p_pos.y / (cell_height * rasterizer_scale_factor)));
+	const int z = static_cast<int>(Math::round(p_pos.z / (cell_size * rasterizer_scale_factor)));
```

Comment:
This `cell_xyz * rasterizer_scale_factor` should be calculated once and cached in some internal var imo.

Changing from floor() to round() will affect existing projects and break compatibility so should be reverted. It also does not change things in a good way as it shifts cells around less predictable when there is high vertex density. To solve merge issues with that there is already the scale factor added.